Deprecate the bfs eauto tactic. (#15314)
coq/coq: doc/sphinx/changes.rst

  (`#15245 <https://github.com/coq/coq/pull/15245>`_,
  fixes `#15244 <https://github.com/coq/coq/issues/15244>`_,
  by Hugo Herbelin).
- **Deprecated:**
  the :tacn:`bfs eauto` tactic. Since its introduction

>   it has behaved exactly like the normal tacn:`eauto` tactic.

Reviewer: ```suggestion
  it has behaved exactly like the normal :tacn:`eauto` tactic.
```

>   it was behaving exactly like the normal tacn:`eauto` tactic.
>   Use instead the `bfs` flag of the :tacn:`typeclass eauto` tactic

Reviewer 2: ```suggestion
  it has behaved exactly like the tacn:`eauto` tactic.
  Use :tacn:`typeclasses eauto` with the `bfs` flag instead
```

Reviewer 3: Why `has behaved` instead of just `behaved`?

Reviewer 3: Or actually `behaves` since it's not removed yet

Reviewer 2: "has behaved" = behavior existed in the past and continues to the present
"behaved" = behavior existed in the past, doesn't say it continues to the present
"behaves" = only talks about the present but "Since its introduction" requires a past tense

If you drop "Since its introduction" then "behaves" is OK.

You could

Author: The present perfect tense in English implies something is still affecting reality. It makes sense to use it there.

Reviewer 2: Also, if it behaves "exactly like the tacn:`eauto` tactic.", why recommend using `typeclasses eauto` instead?

Author: If you want a bfs behaviour, that is.

Reviewer 2: Ah, because it has never worked.
